CMakeLists.txt

```diff
@@ -395,6 +395,7 @@ if(WITH_OPTIM)
             set(ZLIB_ARCH_SRCS ${ZLIB_ARCH_SRCS} ${ARCHDIR}/crc32_acle.c ${ARCHDIR}/insert_string_acle.c)
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ACLEFLAG}")
             add_definitions("-DARM_ACLE_CRC_HASH")
+            add_definitions("-DARM_ACLE_CRC32")
```

Comment:
Using two defines for same purpose is an overkill.

Reply:
But crc32_acle.c  and insert_string_acle.c are for different purpose. Or if you think they should be enabled by the same define, we should merge ARM_ACLE_CRC_HASH and ARM_ACLE_CRC32 into one, like ARM_ACLE ?

Reply:
and change it to : add_definitions("-DARM_ACLE")

Reply:
If you want to keep ACLE enabled for insert_string_acle, make a new define for it instead...

Reply:
I think it's better to use one define for both insert_stream_acle  and crc32_acle, because they both use the functions from arm_acle.h.  And do you think whether "ARM_ACLE"  is a reasonable define for them?

Reply:
I think we should just keep the current define... What comes to functable detection, it will be rewritten soon...